### PR TITLE
MSC3572: Relation aggregation cleanup

### PR DESCRIPTION
**Pull request by bwindels**
[Rendered](https://github.com/matrix-org/matrix-doc/blob/aggregation-cleanup/proposals/3572-aggregation-cleanup.md)

Preview: https://pr3572--matrix-org-previews.netlify.app

[FCP tickyboxes](https://github.com/matrix-org/matrix-spec-proposals/pull/3572#issuecomment-2192866415)

### CODE REVIEW

File: proposals/3572-aggregation-cleanup.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# MSC3572: Relation aggregation cleanup
+
+MSC2575 documented how aggregation have been used in the ecosystem in spite of
```
Comment (uhoreg) [inline proposals/3572-aggregation-cleanup.md:3]: 2575 doesn't look like an MSC -- I think you typed the wrong number here.

File: proposals/3572-aggregation-cleanup.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+them not being part of the spec. This MSC addresses some issues that crept into
+that MSC because the original design of aggregations didn't receive enough feedback.
+
+## Proposal
+
+### Rename `m.relations` in `unsigned` to `relations`
+
+`m.relations` is inconsistent with the other fields under the
+[`unsigned`](https://spec.matrix.org/unstable/client-server-api/#room-event-fields)
```
Comment (clokep) [inline proposals/3572-aggregation-cleanup.md:12]:
```suggestion
[`unsigned`](https://spec.matrix.org/v1.11/client-server-api/#definition-clientevent_unsigneddata)
```

File: proposals/3572-aggregation-cleanup.md (hunk 1 above)
Comment (clokep) [reply proposals/3572-aggregation-cleanup.md:3]:
```suggestion
MSC2675 documented how aggregation have been used in the ecosystem in spite of
```

File: proposals/3572-aggregation-cleanup.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+field.
+
+the CNSIG is really meant for things that are very likely to be extended by
+client/server implementations. Event types are a good example; the contents of
+unsigned is not, really (particularly given everything else in unsigned does not
+use a prefix).
+
+The thing that is particularly objectionable about m.relations is it means that
+a single object (unsigned) contains a mix of namespaced and unnamespaced
+identifiers.
```
Comment (clokep) [inline proposals/3572-aggregation-cleanup.md:22]:
```suggestion
The thing that is particularly objectionable about `m.relations` is it means that
a single object (`unsigned`) contains a mix of namespaced and unnamespaced
identifiers.
```

**Comment by clokep** [comment]
Although I agree it is unfortunate that `unsigned` has a mixed of namespaced and unnamespaced parameters I do not think it is worth the churn to remove the prefix.

Additionally, MSC4115 showed that it is useful to have (unstable) prefixes as an option in `unsigned`.

@mscbot fcp close

**Pull request closed without merging**